instance/models/mixins/storage.py

```diff
@@ -211,14 +212,30 @@ def get_s3_connection(self):
             self.s3_secret_access_key
         )
 
-    def provision_s3(self):
+    def _create_bucket(self, attempts=4, attempt=1, retry_delay=1):
         """
-        Create S3 Bucket if it doesn't exist
+        Create bucket, retry up to defined attempts if it fails
         """
-        if self.s3_access_key and self.s3_secret_access_key:
+        try:
             s3 = self.get_s3_connection()
             bucket = s3.create_bucket(self.s3_bucket_name)
             bucket.set_cors(get_s3_cors_config())
+        except boto.exception.S3ResponseError:
+            if attempts == attempt:
+                raise
+            self.logger.exception(
+                'Retrying bucket creation. IAM keys are not propagated yet, attempt %s of %s',
+                attempt, attempts
+            )
+            time.sleep(retry_delay)
```

Comment:
This is always running in a worker process, never a web process, right? Want to make sure `time.sleep` won't block anything else.

Reply:
Also, if this is known to fail from time to time, perhaps we should always sleep first, to reduce the number of warnings/errors showing up in the logs, instead of expecting to log errors from time to time?

Reply:
agree on adding a sleep first. This runs when a new app server is spawned, so it should be ok.